lib/logger.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var util = require('util');
+var utility = require('./utility');
+var winston = require('winston');
+
+/**
+ * Function to get default options for an implementation of patternlab-import
+ * @return {Object}  options  an object of default patternlab-import options
+ */
+var getDefaultOptions = function (options) {
+  'use strict';
+
+  var options = {
+    logger: {
+      level: 'info',
+      silent: false,
+      exitOnError: true,
+      logFile: ''
+    }
+  };
+
+  return options;
+};
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger.log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} options - Configuration options.
+ * @return {Object} An instance of Winston.
+ */
+exports.logger = function (projectOptions) {
```

Comment:
@e2tha-e 

What calls logger and injects the options?
Would that not trigger the instantiation? Can it be triggered globally?

Options or not, we'd still need to globally instantiate this, correct?